File: code/design-patterns/behavioral/strategy/python/strategy_usage.py
from cache import Cache, FIFO, LRU

cache = Cache(2, LRU())
cache.add("a", 1)
cache.add("b", 2)
cache.add("c", 3)  # This should trigger eviction
cache.set_eviction_algo(FIFO())
cache.add("d", 4)  # This should trigger eviction using FIFO strategy

File: code/design-patterns/behavioral/strategy/go/strategy_usage.go
package main

func main() {
	lru := &LRU{}
	fifo := &FIFO{}
	cache := NewCache(2, lru)

	cache.Add("a", 1)
	cache.Add("b", 2)
	cache.Add("c", 3) // This should trigger eviction with LRU

	cache.SetEvictionAlgo(fifo)
	cache.Add("d", 4) // This should trigger eviction with FIFO
}
